code/game/objects/effects/effect_system/effects_smoke.dm

```diff
@@ -263,8 +263,9 @@
 /datum/effect_system/smoke_spread/chem/New()
 	..()
 	chemholder = new /obj()
-	var/datum/reagents/R = new/datum/reagents(500)
+	var/datum/reagents/R = new/datum/reagents(500, REAGENT_HOLDER_INSTANT_REACT)//This is a safety for now to prevent smoke generating more smoke as the smoke reagents react in the smoke. This is prevented naturally from happening even if this is off, but I want to be sure that any edge cases are prevented before I get a chance to rework smoke reactions (specifically adding water or reacting away stablising agent in the middle of it).
```

Comment:
```suggestion
	var/datum/reagents/R = new (500, REAGENT_HOLDER_INSTANT_REACT) //This is a safety for now to prevent smoke generating more smoke as the smoke reagents react in the smoke. This is prevented naturally from happening even if this is off, but I want to be sure that any edge cases are prevented before I get a chance to rework smoke reactions (specifically adding water or reacting away stabilizing agent in the middle of it).
```
Do you plan to handle this in a future PR?